docs/source/advanced/sagemaker-pipelines.rst

```diff
@@ -0,0 +1,303 @@
+* Track and version your pipeline executions
+
+Pre-requisites
+--------------
+
+Before starting with GraphStorm Pipelines on SageMaker, you'll need:
+
+* Python 3.8 or later
+* An AWS account with appropriate permissions. See the official
+  `SageMaker documentation <https://docs.aws.amazon.com/sagemaker/latest/dg/build-and-manage-access.html>`_
+  for detailed information about required permissions.
+* Basic familiarity with Amazon SageMaker and
+  `SageMaker Pipelines <https://docs.aws.amazon.com/sagemaker/latest/dg/pipelines.html>`_.
+* Understanding of graph neural networks and the `GraphStorm framework <https://graphstorm.readthedocs.io/en/latest/index.html>`_.
+
+Setting Up Your Environment
+---------------------------
+
+To work with GraphStorm SageMaker pipelines, you'll need the GraphStorm source code
+and a Python environment with the SageMaker SDK and AWS SDK (boto3) installed.
+
+1. First, clone the GraphStorm repository and navigate to the pipeline directory:
+
+.. code-block:: bash
+
+   git clone https://github.com/awslabs/graphstorm.git
+   cd graphstorm/sagemaker/pipeline
+
+2. Install the required Python packages:
+
+.. code-block:: bash
+
+   pip install sagemaker boto3
+
+Creating Your First Pipeline
+--------------------------
+
+GraphStorm provides Python scripts to help you create and manage SageMaker pipelines. The main tools are:
+
+* ``create_sm_pipeline.py``: Creates or updates pipeline definitions
+* ``pipeline_parameters.py``: Manages pipeline configuration
+* ``execute_sm_pipeline.py``: Runs pipeline executions
+
+Here's an example of how to create a basic pipeline that includes graph construction, training, and inference:
+
+.. code-block:: bash
+
+    python create_sm_pipeline.py \
+        --graph-construction-config-filename my_gconstruct_config.json \
+        --graph-name my-graph \
+        --graphstorm-pytorch-cpu-image-url 123456789012.dkr.ecr.us-west-2.amazonaws.com/graphstorm:sagemaker-cpu \
+        --input-data-s3 s3://input-bucket/data \
+        --instance-count 2 \
+        --jobs-to-run gconstruct train inference \
+        --output-prefix s3://output-bucket/results \
+        --pipeline-name my-graphstorm-pipeline \
+        --region us-west-2 \
+        --role arn:aws:iam::123456789012:role/SageMakerExecutionRole \
+        --train-inference-task node_classification \
+        --train-yaml-s3 s3://config-bucket/train.yaml
+
+This command sets up a pipeline with three main stages:
+
+1. Graph construction using the configuration in ``my_gconstruct_config.json``
+2. Model training using the settings in ``train.yaml``
+3. Inference using the trained model
+
+The pipeline will use 2 instances (specified by ``--instance-count``) for distributed training and inference.
+
+You'll need to provide:
+
+* A SageMaker execution role (``--role``) with appropriate permissions
+* A GraphStorm Docker image (``--graphstorm-pytorch-cpu-image-url``) for running the tasks
```

Comment:
Can we set different image for graph construction and training/inference?

Reply:
For GSProcessing yes. GConstruct always uses the Pytorch CPU image. Would we like the GConstruct image to be separate?

Reply:
We can set a different image for GSProcessing, GConstruct will use the PyTorch CPU image.

Reply:
OK